hasher-matcher-actioner/hmalib/lambdas/api/api_root.py

```diff
@@ -71,12 +78,72 @@ def root():
 
 def lambda_handler(event, context):
     """
-    root request handler
+    This lambda is invoked in 2 situations:
+
+    1. When the API is called, it uses bottle to process the request and send it to the direct function
+
+    2. Platforms can connect their AWS S3 Buckets directly to HMA so that uploads to those buckets are
+    fed directly into the system. When an upload occurs, this lambda is invoked with an s3 event. We then
+    convert the event into a URL which we submit to the hasher (via SNS)
     """
+    if is_s3_event(event):
+        logger.info(
+            "Lambda triggered with S3 event. Converting to submit content request."
+        )
+        for record in event["Records"]:
```

Comment:
Can you extract this out into a separate function?

the body of `if is_s3_event()`... is too large compared to the `else`.